server/proxy/pf_rdpgfx.c

```diff
@@ -147,9 +147,61 @@ static UINT proxy_MapSurfaceToWindow(RdpgfxClientContext* context,
 	return server->MapSurfaceToWindow(server, surfaceToWindow);
 }
 
+static UINT proxy_OnOpen(RdpgfxClientContext* context, BOOL *do_caps_advertise)
+{
+	WLog_DBG(TAG, "OnOpen");
+	RdpgfxServerContext* server = (RdpgfxServerContext*) context->custom;
+
+	if (NULL != do_caps_advertise)
+		*do_caps_advertise = FALSE;
+
+	printf("pf_rdpgfx: proxy_OnOpen: calling server's Open()\n");
+	// TODO: check if proxy server's dynvc is in ready state
+	// and that client supports gfx, else deinit the channel.
```

Comment:
I don't think that you need to check whether the client supports GFX or not, because this is done in the negotiation

Reply:
i think so too lets remove this comment now

Reply:
but we should check the DVC state ...